### PR TITLE
IgnoreHealthOnHostRemoval Nullable

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Make Upstream.IgnoreHealthOnHostRemoval nullable

# Context

This field was introduced in https://github.com/solo-io/gloo/pull/5325 with a small UDS bug. We then fixed the bug https://github.com/solo-io/gloo/pull/5334 but rushed the fix a little. This PR cleans up the two previous ones, to ensure that we have the nullable field exposed on the upstream, and it is only copied if it is not nil

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/plugins/utils/update_upstream_spec.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func UpdateUpstream(original, desired *v1.Upstream) {
 		}
 	}
 
-	desired.IgnoreHealthOnHostRemoval = original.GetIgnoreHealthOnHostRemoval()
-
+	if desired.GetIgnoreHealthOnHostRemoval() != nil {
```
Comment (kdorosh) [inline projects/gloo/pkg/plugins/utils/update_upstream_spec.go:55]: this should be `==`

**Review by kdorosh — CHANGES_REQUESTED** [review]
small bug

**Comment by sam-heilbron** [comment]
/kick consul data race https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=d3c83d0e-1c23-4d38-a342-e4b2643e8036

**Review by jenshu — APPROVED** [review]

**Review by kdorosh — APPROVED** [review]

**Pull request merged**